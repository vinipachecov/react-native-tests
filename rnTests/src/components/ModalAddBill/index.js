import React, { Component } from 'react';
import {
  Button,
  Modal,
  StyleSheet,
  TextInput,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

class ModalAddBill extends Component {
  state = {
    bill: {},
  };

  render() {
    const { bill } = this.state;
    const { addBill, modalVisible } = this.props;

    return (
      <Modal
        animationType="slide"
        visible={modalVisible}
      >
        <View style={modalStyles.view}>
          <TextInput
            onChangeText={label => this.setState({ bill: { ...bill, label } })}
            placeholder="Bill description (rent, credit card, ...)"
            style={{
              height: 50,
              width: '100%'
            }}
          />
          <TextInput
            keyboard="numeric"
            onChangeText={dueDay => this.setState({ bill: { ...bill, dueDay } })}
            placeholder="Bill due day (08, ...)"
            style={{
              height: 50,
              width: '100%'
            }}
          />
          <TextInput
            keyboard="numeric"
            onChangeText={amount => this.setState({ bill: { ...bill, amount } })}
            placeholder="Bill amount (1500, ...)"
            style={{
              height: 50,
              width: '100%'
            }}
          />
          <Button
            onPress={() => addBill(bill)}
            title="Add bill"
          />
        </View>
      </Modal>
    );
  }
}

ModalAddBill.propTypes = {
  addBill: PropTypes.func.isRequired,
  modalVisible: PropTypes.bool.isRequired,
};

const modalStyles = StyleSheet.create({
  view: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
});

export default ModalAddBill;