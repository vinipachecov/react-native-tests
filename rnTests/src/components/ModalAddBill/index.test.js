import React from 'react';
import { create } from 'react-test-renderer';

import ModalAddBill from './';

describe('[Component] ModalAddBill', () => {

  
  const props = {
    addBill: jest.fn(),
    modalVisible: false    
  };  
  const tree = create(<ModalAddBill { ...props } />);

  it('Matches snapshot', () => {
    expect(tree.toJSON()).toMatchSnapshot();    
  })
  
  
  it('Matches initial state', () => {
    // para pegar o estado é preciso pegar a instancia
    // do componente
    expect(tree.root.instance.state).toEqual({ bill: {} });    
  })

  it('Calls addBill prop', () => {
    // para pegar o estado é preciso pegar a instancia
    // do componente
    // Chamar a função
    tree.root.props.addBill();
    // depois esperar e ver se a função foi chamada
    expect(tree.root.props.addBill).toHaveBeenCalled();    
  })
  
})
