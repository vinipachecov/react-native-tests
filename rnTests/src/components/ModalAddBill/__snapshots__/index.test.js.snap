// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] ModalAddBill Matches snapshot 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Bill description (rent, credit card, ...)"
      style={
        Object {
          "height": 50,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      keyboard="numeric"
      onChangeText={[Function]}
      placeholder="Bill due day (08, ...)"
      style={
        Object {
          "height": 50,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <TextInput
      allowFontScaling={true}
      keyboard="numeric"
      onChangeText={[Function]}
      placeholder="Bill amount (1500, ...)"
      style={
        Object {
          "height": 50,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add bill
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
